### PR TITLE
first pass of responsive component selector

### PR DESCRIPTION
**Pull request by yoshokatana**
* new design
* menu opens/closes on click/tap
* no animations (yet)
* same functionality as before
* [trello ticket](https://trello.com/c/bUZtN6Pa/202-l-component-selector-update)

### CODE REVIEW

File: services/render.js [hunk 1]
```diff
@@ -41,6 +41,14 @@ function addComponentSelector(el) {
   var ref = el.getAttribute(references.referenceAttribute);
 
   return edit.getData(ref)
+    .catch(function (e) {
+      // e.g. tried to get the schema of a component, and it 404'd
```
Comment (yoshokatana) [inline services/render.js]: catching 404s earlier, so they don't get lumped in with other errors

Comment (TakenPilot) [inline services/render.js]: Why returning an object?  Why not null for the lack of an object?

**Comment by TakenPilot** [comment]
Legit.  +1, though I have reservations about return an empty object.  That'll bite us later.

File: services/render.js (hunk 1 above)
Comment (yoshokatana) [inline services/render.js]: fair point

**Comment by TakenPilot** [comment]
That's more obvious, yeah.  +1

**Pull request merged**